src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewModel.swift

```diff
@@ -112,4 +132,15 @@ final class TraceLocationDetailViewModel {
 		let components = Calendar.current.dateComponents([.hour, .minute], from: date ?? Date())
 		return dateComponentsFormatter.string(from: components)
 	}
+	
+	private func generateSHA256(_ guid: String) -> Data? {
+		let encoder = JSONEncoder()
+		do {
+			let guidData = try encoder.encode(guid)
+			return guidData.sha256()
+		} catch {
+			Log.error("traceLocationGUID Encoding error", log: .checkin, error: error)
+			return nil
+		}
+	}
```

Comment:
Why here ant not a utility function/property on `ExposureWindow` so others could use it as well?

Reply:
You mean inside TraceLocations struct ?

Reply:
Yes. This reduces clutter in the view model and the location has a nice property 'hash'.